Next Auth

UZ9 opened the pull request:
## PR Title
Issue Number(s): #34 

What does this PR change and why?
### Checklist
- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [x] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [x] Code follows design and style guidelines
- [x] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes
- Database change / migration to run
- Environment config change
- Breaking API change

### How To Test
The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

UZ9 committed:
:art: Fix blurry wave svgs on login/signup

UZ9 committed:
:sparkles: Add initial Next authentication

afazio1 commented on src/pages/api/auth/[...nextauth].js:
```diff
@@ -1,10 +1,67 @@
 
+let cachedPromise;
+
+import { MongoClient } from "mongodb";
```
remove if not using

afazio1 reviewed (commented):
looks good so far! good work getting the svgs to render properly. Don't forget to add user db docs on notion when ur done with the schema 👍

UZ9 committed:
:hammer: Move DB content generation to User

afazio1 commented on server/db/dbConnect.js:
```diff
@@ -2,6 +2,12 @@ import mongoose from "mongoose";
+  bufferCommands: false,
+};
+
+export const DB_INFO = { DB_URL, DB_NAME, DB_CONNECTION_STRING, DB_OPTS };
```
maybe export the `DB_INFO` variables separately and on an as needed basis. I don't think we need `DB_INFO` as another layer of abstraction for env vars
```js
export const DB_URL
export const DB_NAME
...
```

UZ9 replied:
Done 👍

afazio1 commented on src/pages/api/auth/[...nextauth].js:
```diff
@@ -1,10 +1,100 @@
+import User from "../../../../server/db/models/User";
+import { consts } from "@/utils/consts";
+
+const client = new MongoClient(DB_INFO.DB_CONNECTION_STRING);
```
we only use connection string here, so there's not much of a need to pull in all of `DB_INFO`

UZ9 replied:
Done 👍  (same as above)

afazio1 commented on src/pages/api/auth/[...nextauth].js:
```diff
@@ -1,10 +1,100 @@
+      const _id = new ObjectId(message.user.id);
+
+      // TODO: Find a way of avoiding unnecessary deletion calls
+      await User.deleteOne({ _id });
```
this is not too much of a problem tbh. the operation only happens when new users are created, so not very often

UZ9 replied:
Should I leave the TODO in case we ever come back to it much later?

afazio1 replied:
sure

afazio1 commented on src/utils/consts.js:
```diff
@@ -22,6 +22,7 @@ const consts = {
   concernArray: ["No Concern", "Some Concern", "High Concern"],
   locationArray: ["Facility 1", "Facility 2", "Placed"],
   roleArray: ["Service", "Companion"],
+  userRoleArray: ["Admin", "Instructor", "Volunteer", "Inactive"],
```
leadership decided to change this to 
`["Admin", "Instructor", "Volunteer/Recipient", "Inactive"]`

UZ9 replied:
👍 Done

afazio1 commented on src/pages/api/auth/[...nextauth].js:
```diff
@@ -1,10 +1,100 @@
+        name: message.user.name,
+        email: message.user.email,
+        image: message.user.image,
+        role: "Volunteer" // TODO: Default string shouldn't be hardcoded and instead moved somewhere else
```
make this "Volunteer/Recipient"
- we will probably want to give them a default role of "Inactive" until an admin activates them, but for testing purposes let's leave the default as this

UZ9 replied:
👍 Done

afazio1 reviewed (changes requested):
good job! a few minor fixes and should be good to go

UZ9 committed:
:hammer: PR Changes

- Database info is no longer stored under DB_INFO but file-level
- Removed old imports
- Changed userRole enum to fit specifications
- Changed default role to "Volunteer/Recipient"
- Removed unnecessary method parameters in authentication

UZ9 committed:
:bug: Fix login not redirecting to correct URL

afazio1 committed:
Merge branch 'dev' into ryder-34-next_auth
